great_expectations/expectations/expectation.py

```diff
@@ -2622,7 +2659,9 @@ def _mostly_success(
     unexpected_cnt: int,
     mostly: float,
 ) -> bool:
+    rows_considered_cnt_as_float: float = float(rows_considered_cnt)
+    unexpected_cnt_as_float: float = float(unexpected_cnt)
```

Comment:
There's nothing wrong with this per say, but these kinds of type annotations during assignments are unnecessary.

See these first 2 sections for more details are when this is & isn't needed.
* `Type inference`  
*  `Explicit types for variables`
https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#type-inference-and-type-annotations